e2e/cypress/integration/channel/archived_channels_spec.js

```diff
@@ -55,5 +60,159 @@ describe('Leave an archived channel', () => {
-});
 
+    it('MM-T1670 Can view channel info for an archived channel', () => {
+        // # Visit archived channel
+        cy.visit(`/${testTeam.name}/channels/${testChannel.name}`);
+
+        // # Click the channel header
+        cy.get('#channelHeaderDropdownButton button').click();
+
+        // # Select View Info
+        cy.get('#channelViewInfo button').click();
+
+        // * Channel title is shown with archived icon
+        cy.get('#channelInfoModalLabel span.icon__archive').should('be.visible');
+        cy.contains('#channelInfoModalLabel strong', `${testChannel.display_name}`).should('be.visible');
+
+        // * Channel URL is listed (non-linked text)
+        cy.url().then((loc) => {
+            cy.contains('div.info__value', loc).should('be.visible');
+        });
+    });
+
+    it('MM-T1671 Can view members', () => {
+        // # Open an archived channel
+        cy.visit(`/${testTeam.name}/channels/${testChannel.name}`);
+
+        // # Click the channel header
+        cy.get('#channelHeaderDropdownButton button').click();
+
+        // # Select "View Members"
+        cy.get('#channelViewMembers button').click();
+
+        // * Channel Members modal opens
+        cy.get('div#channelMembersModal').should('be.visible');
+
+        // # Ensure there are no options to change channel roles or membership
+        // * Membership or role cannot be changed
+        cy.get('div[data-testid=userListItemActions]').should('not.be.visible');
+
+        // # Use search box to refine list of members
+        // * Search box works as before to refine member list
+        cy.get('[data-testid=userListItemDetails]').its('length').then((fullListLength) => {
+            cy.get('#searchUsersInput').type(`${testUser.first_name}{enter}`);
+            cy.get('[data-testid=userListItemDetails]').its('length').then((filteredLength) => {
+                expect(fullListLength).to.be.greaterThan(filteredLength);
+            });
+        });
+    });
+    it('MM-T1672_1 User can close archived channel (1/2)', () => {
+        // # Open a channel that's not the town square
+        cy.visit(`/${testTeam.name}/channels/off-topic`);
+
+        // # repeat searching and navigating to the archived channel steps 3 times.
+        [1, 2, 3].forEach((i) => {
+            // * ensure we are not on an archived channel
+            cy.get('#channelInfoModalLabel span.icon__archive').should('not.be.visible');
+
+            // # Search for a post in an archived channel
+            cy.get('#searchBox').focus().clear();
+
+            cy.get('#searchBox').should('be.visible').type(`${testArchivedMessage}{enter}`);
+
+            // # Open the archived channel by selecting Jump from search results and then selecting the link to move to the most recent posts in the channel
+            cy.get('#searchContainer').should('be.visible');
+
+            cy.get('a.search-item__jump').first().click();
+            cy.wait(TIMEOUTS.ONE_SEC); // let the page load
```

Comment:
The waits in these tests don't seem necessary. On the next check, Cypress will wait until that sidebar item is visible, so we shouldn't need an explicit wait.

Reply:
originally there wasn't any, but if the page doesn't fully load, it gets into a race condition where it is still loading but interacting with part of it, so I ended up with ugly waits, I can remove them, but it might fail sometimes.

Reply:
Instead of a wait for a time, you could wait for a certain element to be visible, like I'm waiting in my tests for the sidebar to be open with `cy.get('#sidebarContainer').should('be.visible');`

Reply:
I agree with HH. Also the succeeding ``cy.get(`#sidebarItem_${testChannel.name}`)`` will fail after 20s (far greater than 1s) which is the [default command timeout](https://github.com/mattermost/mattermost-webapp/blob/master/e2e/cypress.json#L4). But I understand that there are some cases where explicit wait is really needed.

Reply:
i'll try to remove them, and see if it now passes (the test has changed a lot since I added them so some parts of it might not fail now)

Reply:
so, I finally figured out what was causing the problem. Once we use the search and jump from the results, we get a permalink, after some time it changes into the actual channel url, since tests sometimes go fast, there was a race condition when I closed the archived channel, sometimes it would have changed the url to the one after close and sometimes the timeout to change the url to the channel would come after.

I've added a check to wait for url change before doing anything else and it seems to have fixed the problem.

Reply:
Yeah, that URL change was causing some issues in my tests as well. I wonder if we'll want to wait for a scroll or something to trigger that since just having it change automatically might make stuff flaky.